api/types/container/config.go

```diff
@@ -13,6 +13,12 @@ import (
 // Docker interprets it as 3 nanoseconds.
 const MinimumDuration = 1 * time.Millisecond
 
+// StopConfig holds the options to stop, or restart a container.
+type StopConfig struct {
+	StopSignal  string `json:",omitempty"` // Signal to stop the container.
+	StopTimeout *int   `json:",omitempty"` // Timeout (in seconds) to stop the container, or '-1' to wait indefinitely.
+}
```

Comment:
🙈 and I just realised _why_ I probably didn't push it as a PR before; I see that I didn't change the client code to allow the new options to be set.

Looking at that, I'm a bit on the fence if `StopConfig` should be used for both `stop` and `restart` (or if separate types should be used).


Let me temporarily move this PR back to draft, and work on the client bits